modules/stochastic_tools/src/vectorpostprocessors/StochasticResults.C

```diff
@@ -75,26 +72,20 @@ StochasticResults::finalize()
 
 void
 StochasticResults::setCurrentLocalVectorPostprocessorValue(
-    const std::string & name, const VectorPostprocessorValue && current)
+    const std::string & vector_name, const VectorPostprocessorValue && current)
```

Comment:
There's not a whole lot of point to const r-value references. https://www.codesynthesis.com/~boris/blog/2012/07/24/const-rvalue-references/

Reply:
Interesting, should it be removed. I just like it because the object might be moving, but the `const` gives me a clue that the function isn't messing with the actual contents.

Reply:
You said the object might be moving... the way this is written it *has* to be moving (or be an rvalue generated in some other way). This method signature won’t accept lvalues. If you know you want  ‘current’ to be const you might as well change this to a const lvalue reference, and then it can accept both lvalues and rvalues. It would give you more flexibility. When I start getting confused by lvalue vs rvalues I like reading this article: https://www.fluentcpp.com/2018/02/06/understanding-lvalues-rvalues-and-their-references/

Reply:
I shouldn't of used "might". This definitely needs to be a move. The "current" vector is accumulated with values when running in batch mode, which is when a single MultiApp instance is re-used for many sub-applications. After all these values are computed in "current" the data is moved into the actual vector postprocessor vector. There can be millions of values in current, so you don't want to copy.

Reply:
An thanks for the link, I am still wrapping my head around lvalue/rvalue stuff.

Reply:
Be careful, "avoiding copying" is not the same as "sharing a reference". What Alex is saying is that you are transferring ownership with an rvalue so the caller doesn't maintain access to that value after the call. This is perfect for some situations (i.e. building up a large structure and then moving it), but may not be appropriate for other uses.

Reply:
> What Alex is saying is that you are transferring ownership with an rvalue so the caller doesn't maintain access to that value after the call.

This is the case here, I am transferring ownership of "current". @permcody You helped me with this when I implemented it a few months ago.

Reply:
[move doesn't move](https://crascit.com/2015/09/21/when-moving-doesnt-move/) :smile: Whether you use an lvalue reference or an rvalue reference here, you're not going to create a copy.

Reply:
Or I should say that you will get just as many copies as you would get using an rvalue reference, e.g. if the compiler performed implicit conversions

Reply:
This code is already there, I just changed the name of a variable. If something needs to be changed to make this code operate differently for performance then I can do that in another PR.

Reply:
I created an issue here, I welcome improvements to the current design: https://github.com/idaholab/moose/issues/15296